src/patroni.py

```diff
@@ -106,3 +145,22 @@ def render_postgresql_conf_file(self) -> None:
             logging_collector="on", synchronous_commit="on", synchronous_standby_names="*"
         )
         self._render_file(f"{self._storage_path}/postgresql-k8s-operator.conf", rendered, 0o644)
+
+    def update_cluster_members(self) -> None:
+        """Update the list of members of the cluster."""
+        # Update the members in the Patroni configuration.
+        self.render_patroni_yml_file()
+
+        try:
+            if self.member_started:
+                # Make Patroni use the updated configuration.
+                self._reload_patroni_configuration()
+        except RetryError:
+            # Ignore retry errors that happen when the member has not started yet.
+            # The configuration will be loaded correctly when Patroni starts.
+            pass
```

Comment:
some functions have `try` `except` on this function for `RetryError`, so you might either want to raise this or remove those `except`s

Reply:
The other functions doesn't have try/except on this function anymore.